test/banach/pagination_test.clj

```diff
@@ -0,0 +1,18 @@
+(ns banach.pagination-test
+  (:require [banach.pagination :as p]
+            [clojure.test :refer [deftest is]]
+            [manifold.deferred :as md]
+            [manifold.stream :as ms]))
+
+(deftest paginated->stream-test
+  (let [gets (atom 0)
+        get! (fn [[marker n]]
+               (is (= ::todo marker))
+               (swap! gets inc)
+               (md/success-deferred
+                {:results (map (partial + (* 10 n)) (range 10))
+                 :todo (when (< n 9) [::todo (inc n)])}))
+        s (p/paginated->stream get! [::todo 0] :todo :results)]
+    (is (< 0 @gets 10))
```

Comment:
I don't think there's a race condition here. The goal of the test is to make sure it's lazy. I don't know exactly how many it fetches ahead of time and I don't care; I just care that it's "more than zero" and "not quite all of them, where all is a sufficiently large number".